theatre/core/src/sheetObjects/TheatreSheetObject.ts

```diff
@@ -153,8 +160,27 @@ export default class TheatreSheetObject<
     return this._valuesDerivation().tapImmediate(coreTicker, fn)
   }
 
+  // internal: Make the deviration keepHot if directly read
   get value(): PropsValue<Props> {
-    return this._valuesDerivation().getValue()
+    const der = this._valuesDerivation()
+    if (KEEP_HOT_FOR_MS != null) {
+      if (!der.isHot && this._keepHotSchedule == null) {
+        const untap = der.keepHot()
+        const clear = () => {
+          if (this._keepHotSchedule?.clear === clear) {
```

Comment:
Not necessarily, because the `_keepHotSchedule` can be re-assigned by a different accessor before `_keepHotSchedule.clear()` is called ()or is something like `setTimeout(_keepHotSchedule.clear, 2000)` is used.

Any ideas for improvement of how I did this are welcome, though.

Reply:
Ah right, thank you for explanation. I think it may be able to simplified (see my other comment). Slightly tricky to reason about this code since the order of execution is not straightforward.